include/OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.hpp

```diff
@@ -0,0 +1,187 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem__
+#define __OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem__
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+
+#include <OpenSpaceToolkit/Physics/Data/Direction.hpp>
+#include <OpenSpaceToolkit/Physics/Data/Scalar.hpp>
+#include <OpenSpaceToolkit/Physics/Unit.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Time.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+namespace system
+{
+
+using ostk::core::types::Real;
+
+using ostk::physics::data::Direction;
+using ostk::physics::data::Scalar;
+using ostk::physics::units::Length;
+using ostk::physics::units::Mass;
+using ostk::physics::units::Time;
+using ostk::physics::units::ElectricCurrent;
+using ostk::physics::units::Angle;
+using ostk::physics::units::Derived;
+using ostk::physics::Unit;
+
+/// @brief                      Define a propulsion system
+
+class PropulsionSystem
+{
+   public:
+    /// @brief              Constructor
+    ///
+    /// @code
+    ///                     PropulsionSystem propulsion = { ... };
+    /// @endcode
+    ///
+    /// @param              [in] aThruster Thruster (scalar)
+    /// @param              [in] aSpecificImpulse Specific impulse
+
+    PropulsionSystem(const Scalar& aThruster, const Scalar& aSpecificImpulse);
+
+    /// @brief              Constructor
+    ///
+    /// @code
+    ///                     PropulsionSystem propulsion = { ... };
+    /// @endcode
+    ///
+    /// @param              [in] aThruster Thruster (real)
+    /// @param              [in] aSpecificImpulse Specific impulse
+
+    PropulsionSystem(const Real& aThruster, const Real& aSpecificImpulse);
```

Comment:
While I like the convenience, I believe we want to only maintain constructors with high level types like Scalar, Mass etc.